### PR TITLE
Add Six Year Anniversary Post

### PR DESCRIPTION
**Pull request by XAMPPRocky**
### [Rendered](https://github.com/XAMPPRocky/blog.rust-lang.org/blob/patch-1/posts/2021-05-15-six-years-of-rust.md)

r? @rust-lang/core

### CODE REVIEW

File: posts/2020-05-15-six-years-of-rust.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+---
+layout: post
+title: "Six Years of Rust"
+author: The Rust Core Team
```
Comment (Manishearth) [inline posts/2020-05-15-six-years-of-rust.md:4]: I would probably say "The Rust Team" here

File: posts/2020-05-15-six-years-of-rust.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+---
+
+Today marks [Rust]'s sixth birthday since it went 1.0 in 2015. A lot has changed since then and especially over the past year, and Rust was no different. In 2020, there was no foundation yet, no const generics, and a lot organisations were still wondering whether Rust was production ready. 
+
+In the midst of the COVID-19 pandemic, hundreds of Rust's global distributed set of team members and volunteers shipped over 9 major releases of Rust and multiple patch releases. Today, "Rust in production" isn't a question, but a statement. The newly founded Rust foundation's has several members who value using Rust in production enough to help continue to support and contribute to its open development ecosystem.
```
Comment (joshtriplett) [inline posts/2020-05-15-six-years-of-rust.md:9]:
```suggestion
In the midst of the COVID-19 pandemic, hundreds of Rust's global distributed set of team members and volunteers shipped over 9 major releases of Rust and multiple patch releases. Today, "Rust in production" isn't a question, but a statement. The newly founded Rust foundation has several members who value using Rust in production enough to help continue to support and contribute to its open development ecosystem.
```

Another option would be something like "foundation's membership includes", with corresponding tweaks to the rest of the sentence.

Comment (Mark-Simulacrum) [inline posts/2020-05-15-six-years-of-rust.md:9]: nit: we shipped minor releases in the semver sense - probably makes sense to find a non-semver-used word here.

File: posts/2020-05-15-six-years-of-rust.md (hunk 1 above)
Comment (XAMPPRocky) [reply posts/2020-05-15-six-years-of-rust.md:4]: It's been attributed to the core team in previous years, but I have no opinion myself.

File: posts/2020-05-15-six-years-of-rust.md (hunk 2 above)
Comment (XAMPPRocky) [reply posts/2020-05-15-six-years-of-rust.md]: Yeah, I did notice that and wasn't really happy with it, but couldn't come up with a better alternative. Happy to hear  alternatives though.

File: posts/2020-05-15-six-years-of-rust.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+
+## Rust In Production
+Each year Rust's growth and adoption in the community and industry has been unbelievable, and this past year has been no exception. Once again for the sixth year in a row, Rust has been voted Stackoverflow's Most Loved Programming Language.
+
+With the formation of the [Rust foundation], Rust has been in a better position build a sustainable open source ecosystem empowering everyone to build reliable and efficient software.
+
+And it isn't just Rust that has been getting bigger. Larger and larger companies have been adopting Rust in their projects and offering officially supported Rust APIs.
+
+- Both Microsoft and Amazon have just recently announced and released their new officially supported Rust libraries for interacting with [Windows] and [AWS]. Official first party support for these massive and monolithic APIs helps make Rust people's first choice when deciding what to use for their project.
```
Comment (joshtriplett) [inline posts/2020-05-15-six-years-of-rust.md:38]:
```suggestion
- Both Microsoft and Amazon have just recently announced and released their new officially supported Rust libraries for interacting with [Windows] and [AWS]. Official first party support for these massive APIs helps make Rust people's first choice when deciding what to use for their project.
```

I don't think "monolithic" is the right term here' they're quite modular.

File: posts/2020-05-15-six-years-of-rust.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+---
+
+Are there changes, or projects from the past year that you're excited about? Are you looking to get started with Rust? Do you want to help contribute to 2021? Then come, introduce yourself, join the discussion over on our [Discourse] forum and [Zulip] chat! Everyone is welcome, and we are committed to providing a friendly, safe and welcoming environment for all, regardless of gender, sexual orientation, disability, ethnicity, religion, or similar personal characteristic.
+
+[ferrocene]: https://ferrous-systems.com/ferrocene
+[`rust-gpu`]: https://github.com/EmbarkStudios/rust-gpu
+[linux-rust]: https://lore.kernel.org/lkml/CANiq72khBa2GcB6-PHM3A44Y90d6vzYAS=BVpk3nT4B6u+NVDw@mail.gmail.com/T/#mb5e524dae9d5a5815c6e68eb36b9bde4e87c861d
+[edition-plan]: https://blog.rust-lang.org/2021/05/11/edition-2021.html
+[discourse]: TODO
```
Comment (joshtriplett) [inline posts/2020-05-15-six-years-of-rust.md:69]:
```suggestion
[discourse]: https://users.rust-lang.org/
```

File: posts/2020-05-15-six-years-of-rust.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+[tls]: https://daniel.haxx.se/blog/2021/02/09/curl-supports-rustls/
+[rust foundation]: https://foundation.rust-lang.org/posts/2021-02-08-hello-world/
+[windows]:https://github.com/microsoft/windows-rs
+[aws]: https://github.com/awslabs/aws-sdk-rust
+
+## Future Work
+Of course, all that is just to start, we're seeing more and more initiatives putting Rust in exciting new places;
+
+- Critical Section & Ferrous Systems have started [Ferrocene]. A project to make Rust a viable programming language for safety and mission critical systems across the industry.
+- Embark Studios have released an initial prototype of [`rust-gpu`], a new compiler backend that allows writing graphics shaders using Rust for GPUs.
```
Comment (joshtriplett) [inline posts/2020-05-15-six-years-of-rust.md:53]:
```suggestion
- Embark Studios have released an initial prototype of [`rust-gpu`], a new compiler backend that allows writing graphics shaders using Rust for GPUs, and is now [using it in production for 100% of their shaders](https://mailchi.mp/embark-studios/dev009)
```

(Feel free to split the link to a footnote, something like "rust-gpu-in-production".)

File: posts/2020-05-15-six-years-of-rust.md (hunk 2 above)
Comment (joshtriplett) [reply posts/2020-05-15-six-years-of-rust.md]: Perhaps "nine new stable releases of Rust, in addition to various bugfix releases"?

File: posts/2020-05-15-six-years-of-rust.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+-  On supporting an even wider ecosystem, the introduction of tier 1 support for 64 bit ARM Linux, and tier 2 support for ARM macOS & ARM Windows, has made Rust an even better place to easily build your projects across new and different architectures.
+
+-  The most notable exception to the theme of polish has been the major improvements to Rust's compile-time capabilities. The stabilisation of const generics for primitive types, the addition of control flow for `const fn`s, and allowing procedural macros to be used in more places, have allowed completely powerful new types of APIs and crates to be created.
+
+Rustc wasn't the only tool that had significant improvements.
+
+- Cargo just recently stabilised its new feature resolver, that makes it easier to use your dependencies across different targets.
+
+- Rustdoc stabilised its "intra-doc links" feature, allowing you to easily and automatically cross reference Rust types and functions in your documentation.
+
```
Comment (Mark-Simulacrum) [inline posts/2020-05-15-six-years-of-rust.md:29]: We can highlight clippy & cargo learning to rerun properly and print output after cargo check.

File: posts/2020-05-15-six-years-of-rust.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+## Rust In Production
+Each year Rust's growth and adoption in the community and industry has been unbelievable, and this past year has been no exception. Once again for the sixth year in a row, Rust has been voted Stackoverflow's Most Loved Programming Language.
```
Comment (Mark-Simulacrum) [inline posts/2020-05-15-six-years-of-rust.md:32]:
```suggestion
Each year Rust's growth and adoption in the community and industry has been unbelievable, and this past year has been no exception. Once again for the sixth year in a row, Rust has been voted StackOverflow's Most Loved Programming Language.
```

I think? May be worth linking, too.

File: posts/2020-05-15-six-years-of-rust.md (hunk 1 above)
Comment (steveklabnik) [reply posts/2020-05-15-six-years-of-rust.md:4]: The core team didn't write this post, and so we shouldn't be named as the authors.

File: posts/2020-05-15-six-years-of-rust.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
+---
+layout: post
+title: "Six Years of Rust"
+author: The Rust Core Team
+---
+
+Today marks [Rust]'s sixth birthday since it went 1.0 in 2015. A lot has changed since then and especially over the past year, and Rust was no different. In 2020, there was no foundation yet, no const generics, and a lot organisations were still wondering whether Rust was production ready. 
+
+In the midst of the COVID-19 pandemic, hundreds of Rust's global distributed set of team members and volunteers shipped over 9 major releases of Rust and multiple patch releases. Today, "Rust in production" isn't a question, but a statement. The newly founded Rust foundation has several members who value using Rust in production enough to help continue to support and contribute to its open development ecosystem.
```
Comment (steveklabnik) [inline posts/2020-05-15-six-years-of-rust.md:9]:
```suggestion
In the midst of the COVID-19 pandemic, hundreds of Rust's global distributed set of team members and volunteers shipped over nine major releases of Rust and multiple patch releases. Today, "Rust in production" isn't a question, but a statement. The newly founded Rust foundation has several members who value using Rust in production enough to help continue to support and contribute to its open development ecosystem.
```

extremely tiny nit: usually numbers less than ten are written out

File: posts/2020-05-15-six-years-of-rust.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+[rust]: https://www.rust-lang.org
+
+## Recent Additions
+The Rust language has improved tremendously in the past year, gaining a lot of quality of life features, that while they don't fundamentally change the language, they help make using and maintaining Rust in more places even easier.
+
+-  As of Rust 1.52.0 and the upgrade to LLVM 12, one of few cases of unsoundness around forward progress (such as handling infinite loops) has finally been resolved. This has been a long running collaboration between the Rust teams and the LLVM project, and is a great example of improvements to Rust also benefitting the wider ecosystem of programming languages.
+
+-  On supporting an even wider ecosystem, the introduction of tier 1 support for 64 bit ARM Linux, and tier 2 support for ARM macOS & ARM Windows, has made Rust an even better place to easily build your projects across new and different architectures.
```
Comment (steveklabnik) [inline posts/2020-05-15-six-years-of-rust.md:20]: Do we capitalize `Tier` here? I think we do? Not sure.

File: posts/2020-05-15-six-years-of-rust.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+
+
+## Rust In Production
+Each year Rust's growth and adoption in the community and industry has been unbelievable, and this past year has been no exception. Once again for the sixth year in a row, Rust has been voted Stackoverflow's Most Loved Programming Language.
+
+With the formation of the [Rust foundation], Rust has been in a better position build a sustainable open source ecosystem empowering everyone to build reliable and efficient software.
+
+And it isn't just Rust that has been getting bigger. Larger and larger companies have been adopting Rust in their projects and offering officially supported Rust APIs.
+
+- Both Microsoft and Amazon have just recently announced and released their new officially supported Rust libraries for interacting with [Windows] and [AWS]. Official first party support for these massive APIs helps make Rust people's first choice when deciding what to use for their project.
```
Comment (steveklabnik) [inline posts/2020-05-15-six-years-of-rust.md:38]: Worth mentioning Google and Android here too?

Comment (steveklabnik) [reply posts/2020-05-15-six-years-of-rust.md:38]: I guess that's below in "future" work. I thought some of it was shipped currently. Seems fine.

**Review by steveklabnik — APPROVED** [review]
Thanks for putting this together.

File: posts/2020-05-15-six-years-of-rust.md (hunk 8 above)
Comment (XAMPPRocky) [reply posts/2020-05-15-six-years-of-rust.md]: Always welcome, It's my favourite nit. :)

File: posts/2020-05-15-six-years-of-rust.md (hunk 10 above)
Comment (XAMPPRocky) [reply posts/2020-05-15-six-years-of-rust.md:38]: Yeah, unless there's an other official google projects that are further along worth mentioning? I don't keep too much track of Fuchsia myself.

File: posts/2020-05-15-six-years-of-rust.md (hunk 5 above)
Comment (joshtriplett) [reply posts/2020-05-15-six-years-of-rust.md:53]: (Closing this based on discussion with Embark folks.)

File: posts/2020-05-15-six-years-of-rust.md [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+
+- Cargo just recently stabilised its new feature resolver, that makes it easier to use your dependencies across different targets.
+
+- Rustdoc stabilised its "intra-doc links" feature, allowing you to easily and automatically cross reference Rust types and functions in your documentation.
+
+
+## Rust In Production
+Each year Rust's growth and adoption in the community and industry has been unbelievable, and this past year has been no exception. Once again for the sixth year in a row, Rust has been voted StackOverflow's Most Loved Programming Language.
+
+With the formation of the [Rust foundation], Rust has been in a better position build a sustainable open source ecosystem empowering everyone to build reliable and efficient software.
```
Comment (nikomatsakis) [inline posts/2020-05-15-six-years-of-rust.md:34]:
```suggestion
With the formation of the [Rust foundation], Rust has been in a better position build a sustainable open source ecosystem empowering everyone to build reliable and efficient software. Further, A number of companies that use Rust have formed teams dedicated to maintaining and improving the Rust project, including [AWS](https://aws.amazon.com/blogs/opensource/how-our-aws-rust-team-will-contribute-to-rusts-future-successes/), [Facebook](https://engineering.fb.com/2021/04/29/developer-tools/rust/), and Microsoft.
```

Comment (nikomatsakis) [reply posts/2020-05-15-six-years-of-rust.md:34]: I'm not sure what to link to for Microsoft (cc @wesleywiser) -- and maybe I'm forgetting a team somewhere?

**Review by nikomatsakis — DISMISSED** [review]
This looks great! Thanks for doing this, @XAMPPRocky

File: posts/2020-05-15-six-years-of-rust.md (hunk 9 above)
Comment (nikomatsakis) [reply posts/2020-05-15-six-years-of-rust.md:20]: It looked funny to me too :)

File: posts/2020-05-15-six-years-of-rust.md (hunk 11 above)
Comment (wesleywiser) [reply posts/2020-05-15-six-years-of-rust.md:34]: Unfortunately, we don't have anything to link to at the moment, but thanks for checking!

**Comment by programmerjake** [comment]
shouldn't the file name have 2021 instead of 2020?

**Comment by XAMPPRocky** [comment]
@programmerjake Is it 2021 already? 🙃 Good catch!

**Review by Mark-Simulacrum — APPROVED** [review]

**Pull request merged**